// @mantine/core/cjs/components/Slider/Marks/Marks.styles.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var styles = require('@mantine/styles');
var SliderRoot_styles = require('../SliderRoot/SliderRoot.styles.js');

var useStyles = styles.createStyles((theme, { size, color }) => ({
  markWrapper: {
    position: "absolute",
    top: 0
  },
  mark: {
    boxSizing: "border-box",
    border: `${styles.getSizeValue({ size, sizes: SliderRoot_styles.sizes }) >= 8 ? "2px" : "1px"} solid ${theme.colorScheme === "dark" ? theme.colors.dark[4] : theme.colors.gray[2]}`,
    zIndex: 1,
    height: styles.getSizeValue({ sizes: SliderRoot_styles.sizes, size }),
    width: styles.getSizeValue({ sizes: SliderRoot_styles.sizes, size }),
    borderRadius: 1e3,
    transform: `translateX(-${styles.getSizeValue({ sizes: SliderRoot_styles.sizes, size }) / 2}px)`,
    backgroundColor: theme.colorScheme === "dark" ? theme.colors.dark[0] : theme.white
  },
  markFilled: {
    borderColor: styles.getThemeColor({ theme, color, shade: 6 })
  },
  markLabel: {
    transform: "translate(-50%, 0)",
    fontSize: theme.fontSizes.sm,
    color: theme.colorScheme === "dark" ? theme.colors.dark[2] : theme.colors.gray[6],
    marginTop: theme.spacing.xs / 2
  }
}));

exports.default = useStyles;
