// @mantine/core/esm/components/Accordion/AccordionItem/AccordionItem.js
import React from 'react';
import { useForceUpdate, useReducedMotion, useWindowEvent } from '@mantine/hooks';
import { useSx } from '@mantine/styles';
import { ChevronIcon } from './ChevronIcon.js';
import useStyles from './AccordionItem.styles.js';
import { UnstyledButton } from '../../Button/UnstyledButton/UnstyledButton.js';
import { Center } from '../../Center/Center.js';
import { Collapse } from '../../Collapse/Collapse.js';

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
function AccordionItem(_a) {
  var _b = _a, {
    opened,
    onToggle,
    label,
    children,
    className,
    classNames,
    styles,
    sx,
    transitionDuration,
    icon = /* @__PURE__ */ React.createElement(ChevronIcon, null),
    disableIconRotation = false,
    iconPosition = "left",
    id
  } = _b, others = __objRest(_b, [
    "opened",
    "onToggle",
    "label",
    "children",
    "className",
    "classNames",
    "styles",
    "sx",
    "transitionDuration",
    "icon",
    "disableIconRotation",
    "iconPosition",
    "id"
  ]);
  const forceUpdate = useForceUpdate();
  const reduceMotion = useReducedMotion();
  const duration = reduceMotion ? 0 : transitionDuration;
  const { sxClassName } = useSx({ sx });
  const { classes, cx } = useStyles({ transitionDuration: duration, disableIconRotation, iconPosition }, { classNames, styles, name: "Accordion" });
  useWindowEvent("resize", () => forceUpdate());
  return /* @__PURE__ */ React.createElement("div", __spreadValues({
    className: cx(classes.item, { [classes.itemOpened]: opened }, sxClassName, className)
  }, others), /* @__PURE__ */ React.createElement(UnstyledButton, {
    className: classes.control,
    onClick: onToggle,
    type: "button",
    "aria-expanded": opened,
    "aria-controls": `${id}-body`,
    id
  }, /* @__PURE__ */ React.createElement(Center, {
    className: classes.icon
  }, icon), /* @__PURE__ */ React.createElement("div", {
    className: classes.label
  }, label)), /* @__PURE__ */ React.createElement(Collapse, {
    in: opened,
    transitionDuration: duration
  }, /* @__PURE__ */ React.createElement("div", {
    className: classes.content,
    role: "region",
    id: `${id}-body`,
    "aria-labelledby": id
  }, /* @__PURE__ */ React.createElement("div", {
    className: classes.contentInner
  }, children))));
}
AccordionItem.displayName = "@mantine/core/AccordionItem";

export { AccordionItem };
