// @mantine/core/esm/components/Popover/PopoverBody/PopoverBody.styles.js
import { createStyles, getFocusStyles, getSizeValue } from '@mantine/styles';

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var useStyles = createStyles((theme, { radius, shadow, spacing }) => ({
  title: {},
  wrapper: __spreadValues({}, getFocusStyles(theme)),
  popover: {
    background: theme.colorScheme === "dark" ? theme.colors.dark[6] : theme.white,
    pointerEvents: "all",
    borderRadius: getSizeValue({ size: radius, sizes: theme.radius })
  },
  body: {
    border: `1px solid ${theme.colorScheme === "dark" ? theme.colors.dark[6] : theme.colors.gray[2]}`,
    boxShadow: shadow in theme.shadows ? theme.shadows[shadow] : shadow,
    borderRadius: getSizeValue({ size: radius, sizes: theme.radius })
  },
  inner: {
    padding: getSizeValue({ size: spacing, sizes: theme.spacing })
  },
  header: {
    borderBottom: `1px solid ${theme.colorScheme === "dark" ? theme.colors.dark[7] : theme.colors.gray[2]}`,
    padding: `${theme.spacing.xs / 1.5}px ${getSizeValue({
      size: spacing,
      sizes: theme.spacing
    })}px`,
    display: "flex",
    alignItems: "center",
    justifyContent: "space-between"
  },
  close: {
    position: "absolute",
    top: 7,
    zIndex: 2,
    right: 10
  }
}));

export default useStyles;
