// @mantine/core/esm/components/Mark/Mark.js
import React, { forwardRef } from 'react';
import { useMantineTheme, useSx, useExtractedMargins, getThemeColor } from '@mantine/styles';

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const Mark = forwardRef((_a, ref) => {
  var _b = _a, { color = "yellow", style, sx, className } = _b, others = __objRest(_b, ["color", "style", "sx", "className"]);
  const theme = useMantineTheme();
  const { sxClassName, css, cx } = useSx({ sx, className });
  const { mergedStyles, rest } = useExtractedMargins({ others, style });
  return /* @__PURE__ */ React.createElement("mark", __spreadValues({
    ref,
    style: mergedStyles,
    className: cx(css({
      backgroundColor: getThemeColor({
        theme,
        color,
        shade: theme.colorScheme === "dark" ? 5 : 2
      }),
      color: theme.colorScheme === "dark" ? theme.colors.dark[9] : "inherit"
    }), sxClassName)
  }, rest));
});
Mark.displayName = "@mantine/core/Mark";

export { Mark };
