// @mantine/core/esm/components/Timeline/Timeline.js
import React, { Children } from 'react';
import { useSx, useExtractedMargins } from '@mantine/styles';
import { TimelineItem } from './TimelineItem/TimelineItem.js';

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
function Timeline(_a) {
  var _b = _a, {
    children,
    style,
    active,
    color,
    bulletSize = 20,
    align = "left",
    lineWidth = 4,
    classNames,
    styles,
    className,
    sx
  } = _b, others = __objRest(_b, [
    "children",
    "style",
    "active",
    "color",
    "bulletSize",
    "align",
    "lineWidth",
    "classNames",
    "styles",
    "className",
    "sx"
  ]);
  const { sxClassName } = useSx({ sx, className });
  const hasActive = typeof active === "number";
  const { mergedStyles, rest } = useExtractedMargins({ others, style });
  const items = Children.toArray(children).filter((child) => child.type === TimelineItem).map((item, index) => React.cloneElement(item, {
    classNames,
    styles,
    align,
    lineWidth,
    color: item.props.color || color,
    bulletSize: item.props.bulletSize || bulletSize,
    active: item.props.active || hasActive && active >= index,
    lineActive: item.props.lineActive || hasActive && active - 1 >= index
  }));
  const offset = align === "left" ? { paddingLeft: bulletSize / 2 + lineWidth / 2 } : { paddingRight: bulletSize / 2 + lineWidth / 2 };
  return /* @__PURE__ */ React.createElement("div", __spreadValues({
    className: sxClassName,
    style: __spreadValues(__spreadValues({}, offset), mergedStyles)
  }, rest), items);
}
Timeline.Item = TimelineItem;
Timeline.displayName = "@mantine/core/Timeline";

export { Timeline };
