// @mantine/core/cjs/components/Accordion/AccordionItem/AccordionItem.styles.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var styles = require('@mantine/styles');

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
const ICON_SIZE = 24;
var useStyles = styles.createStyles((theme, { transitionDuration, disableIconRotation, iconPosition }, getRef) => {
  const icon = {
    ref: getRef("icon"),
    transition: `transform ${transitionDuration}ms ease`,
    marginRight: iconPosition === "right" ? 0 : theme.spacing.sm,
    marginLeft: iconPosition === "right" ? theme.spacing.lg : 0,
    width: ICON_SIZE,
    minWidth: ICON_SIZE,
    height: ICON_SIZE,
    borderRadius: ICON_SIZE
  };
  return {
    icon,
    label: {
      color: "inherit",
      fontWeight: 500,
      flex: 1,
      overflow: "hidden",
      textOverflow: "ellipsis"
    },
    item: {
      color: theme.colorScheme === "dark" ? theme.colors.dark[0] : theme.black,
      borderBottom: `1px solid ${theme.colorScheme === "dark" ? theme.colors.dark[4] : theme.colors.gray[3]}`
    },
    itemOpened: {
      [`& .${icon.ref}`]: {
        transform: disableIconRotation ? "none" : "rotate(180deg)"
      }
    },
    control: __spreadProps(__spreadValues(__spreadValues({}, styles.getFocusStyles(theme)), styles.getFontStyles(theme)), {
      width: "100%",
      display: "flex",
      alignItems: "center",
      flexDirection: iconPosition === "right" ? "row-reverse" : "row",
      padding: `${theme.spacing.sm}px ${theme.spacing.md / 2}px`,
      paddingLeft: iconPosition === "right" ? theme.spacing.sm + 4 : null,
      fontWeight: 500,
      textAlign: "left",
      color: theme.colorScheme === "dark" ? theme.colors.dark[0] : theme.black,
      "&:hover": {
        backgroundColor: theme.colorScheme === "dark" ? theme.colors.dark[5] : theme.colors.gray[0]
      }
    }),
    content: __spreadProps(__spreadValues({}, styles.getFontStyles(theme)), {
      lineHeight: theme.lineHeight,
      paddingLeft: iconPosition === "right" ? 0 : ICON_SIZE + theme.spacing.xs / 2
    }),
    contentInner: {
      padding: theme.spacing.md,
      paddingTop: theme.spacing.xs / 2
    }
  };
});

exports.default = useStyles;
