// @mantine/core/cjs/components/Loader/Loader.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var styles = require('@mantine/styles');
var Bars = require('./loaders/Bars.js');
var Oval = require('./loaders/Oval.js');
var Dots = require('./loaders/Dots.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e['default'] : e; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const LOADERS = {
  bars: Bars.Bars,
  oval: Oval.Oval,
  dots: Dots.Dots
};
const LOADER_SIZES = {
  xs: 18,
  sm: 22,
  md: 36,
  lg: 44,
  xl: 58
};
function Loader(_a) {
  var _b = _a, {
    size = "md",
    color,
    className,
    variant,
    style,
    sx
  } = _b, others = __objRest(_b, [
    "size",
    "color",
    "className",
    "variant",
    "style",
    "sx"
  ]);
  const { sxClassName } = styles.useSx({ sx, className });
  const theme = styles.useMantineTheme();
  const { mergedStyles, rest } = styles.useExtractedMargins({ others, style });
  const defaultLoader = variant in LOADERS ? variant : theme.loader;
  const Component = LOADERS[defaultLoader] || LOADERS.bars;
  const _color = color || theme.primaryColor;
  return /* @__PURE__ */ React__default.createElement(Component, __spreadValues({
    size: styles.getSizeValue({ size, sizes: LOADER_SIZES }),
    style: mergedStyles,
    color: _color in theme.colors ? styles.getThemeColor({ theme, color: _color, shade: theme.colorScheme === "dark" ? 4 : 6 }) : color,
    role: "presentation",
    className: sxClassName
  }, rest));
}
Loader.displayName = "@mantine/core/Loader";

exports.LOADER_SIZES = LOADER_SIZES;
exports.Loader = Loader;
