// @mantine/styles/cjs/theme/MantineProvider.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var defaultTheme = require('./default-theme.js');
var mergeTheme = require('./utils/merge-theme/merge-theme.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e['default'] : e; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

const MantineThemeContext = React.createContext({
  theme: defaultTheme.DEFAULT_THEME,
  styles: {},
  emotionOptions: { key: "mantine", prepend: true }
});
function useMantineTheme() {
  var _a;
  return ((_a = React.useContext(MantineThemeContext)) == null ? void 0 : _a.theme) || defaultTheme.DEFAULT_THEME;
}
function useMantineThemeStyles() {
  var _a;
  return ((_a = React.useContext(MantineThemeContext)) == null ? void 0 : _a.styles) || {};
}
function useMantineEmotionOptions() {
  var _a;
  return ((_a = React.useContext(MantineThemeContext)) == null ? void 0 : _a.emotionOptions) || { key: "mantine", prepend: true };
}
function MantineProvider({
  theme,
  styles = {},
  emotionOptions,
  children
}) {
  return /* @__PURE__ */ React__default.createElement(MantineThemeContext.Provider, {
    value: { theme: mergeTheme.mergeTheme(defaultTheme.DEFAULT_THEME, theme), styles, emotionOptions }
  }, children);
}
MantineProvider.displayName = "@mantine/core/MantineProvider";

exports.MantineProvider = MantineProvider;
exports.useMantineEmotionOptions = useMantineEmotionOptions;
exports.useMantineTheme = useMantineTheme;
exports.useMantineThemeStyles = useMantineThemeStyles;
