// @mantine/core/cjs/components/Group/Group.styles.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var styles = require('@mantine/styles');

const POSITIONS = {
  left: "flex-start",
  center: "center",
  right: "flex-end",
  apart: "space-between"
};
var useStyles = styles.createStyles((theme, { spacing, position, noWrap, direction, grow, align, withGutter, count }) => ({
  root: {
    display: "flex",
    flexDirection: direction,
    alignItems: align || (direction === "row" ? "center" : grow ? "stretch" : POSITIONS[position]),
    flexWrap: noWrap ? "nowrap" : "wrap",
    justifyContent: direction === "row" && POSITIONS[position],
    margin: -1 * styles.getSizeValue({ size: spacing, sizes: theme.spacing }) / 2,
    padding: withGutter ? styles.getSizeValue({ size: spacing, sizes: theme.spacing }) / 2 : 0
  },
  child: {
    maxWidth: grow && direction === "row" ? `calc(${100 / count}% - ${styles.getSizeValue({ size: spacing, sizes: theme.spacing })}px)` : void 0,
    flexGrow: grow ? 1 : 0,
    margin: styles.getSizeValue({ size: spacing, sizes: theme.spacing }) / 2
  }
}));

exports.default = useStyles;
