// @mantine/core/cjs/components/Overlay/Overlay.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var styles = require('@mantine/styles');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e['default'] : e; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const Overlay = React.forwardRef((_a, ref) => {
  var _b = _a, {
    className,
    opacity = 0.6,
    color = "#fff",
    gradient,
    zIndex = 1e3,
    component,
    radius = 0,
    sx
  } = _b, others = __objRest(_b, [
    "className",
    "opacity",
    "color",
    "gradient",
    "zIndex",
    "component",
    "radius",
    "sx"
  ]);
  const { css, sxClassName, cx } = styles.useSx({ sx, className });
  const theme = styles.useMantineTheme();
  const Element = component || "div";
  const background = gradient ? { backgroundImage: gradient } : { backgroundColor: color };
  return /* @__PURE__ */ React__default.createElement(Element, __spreadValues({
    className: cx(css(__spreadProps(__spreadValues({}, background), {
      opacity,
      position: "absolute",
      top: 0,
      bottom: 0,
      left: 0,
      right: 0,
      borderRadius: styles.getSizeValue({ size: radius, sizes: theme.radius }),
      zIndex
    })), sxClassName),
    ref
  }, others));
});
Overlay.displayName = "@mantine/core/Overlay";

exports.Overlay = Overlay;
