// @mantine/core/cjs/components/Collapse/Collapse.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var hooks = require('@mantine/hooks');
var styles = require('@mantine/styles');
var useCollapse = require('./use-collapse.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e['default'] : e; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
function Collapse(_a) {
  var _b = _a, {
    children,
    in: opened,
    transitionDuration = 200,
    transitionTimingFunction = "ease",
    style,
    sx,
    className,
    onTransitionEnd,
    animateOpacity = true
  } = _b, others = __objRest(_b, [
    "children",
    "in",
    "transitionDuration",
    "transitionTimingFunction",
    "style",
    "sx",
    "className",
    "onTransitionEnd",
    "animateOpacity"
  ]);
  const reduceMotion = hooks.useReducedMotion();
  const duration = reduceMotion ? 0 : transitionDuration;
  const { mergedStyles, rest } = styles.useExtractedMargins({ others, style });
  const { sxClassName } = styles.useSx({ sx, className });
  const getCollapseProps = useCollapse.useCollapse({
    opened,
    transitionDuration: duration,
    transitionTimingFunction,
    onTransitionEnd
  });
  return /* @__PURE__ */ React__default.createElement("div", __spreadProps(__spreadValues({}, getCollapseProps(__spreadValues({ style: mergedStyles }, rest))), {
    className: sxClassName
  }), /* @__PURE__ */ React__default.createElement("div", {
    style: {
      opacity: opened ? 1 : 0,
      transition: animateOpacity ? `opacity ${duration}ms ${transitionTimingFunction}` : "none"
    }
  }, children));
}
Collapse.displayName = "@mantine/core/Collapse";

exports.Collapse = Collapse;
