// @mantine/styles/cjs/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defaultTheme = require('./theme/default-theme.js');
var MantineProvider = require('./theme/MantineProvider.js');
var ColorSchemeProvider = require('./theme/ColorSchemeProvider.js');
var GlobalStyles = require('./theme/GlobalStyles.js');
var NormalizeCSS = require('./theme/NormalizeCSS.js');
var getFocusStyles = require('./theme/utils/get-focus-styles/get-focus-styles.js');
var getFontStyles = require('./theme/utils/get-font-styles/get-font-styles.js');
var getSharedColorScheme = require('./theme/utils/get-shared-color-scheme/get-shared-color-scheme.js');
var getSizeValue = require('./theme/utils/get-size-value/get-size-value.js');
var getThemeColor = require('./theme/utils/get-theme-color/get-theme-color.js');
var hexToRgba = require('./theme/utils/hex-to-rgba/hex-to-rgba.js');
var useExtractedMargins = require('./theme/utils/use-extracted-margins/use-extracted-margins.js');
var useSx = require('./theme/utils/use-sx/use-sx.js');
var react = require('@emotion/react');
var createStyles = require('./tss/create-styles.js');
var Global = require('./tss/Global.js');
var CacheProvider = require('./tss/CacheProvider.js');
var useCss = require('./tss/use-css.js');



exports.DEFAULT_THEME = defaultTheme.DEFAULT_THEME;
exports.MANTINE_COLORS = defaultTheme.MANTINE_COLORS;
exports.MANTINE_SIZES = defaultTheme.MANTINE_SIZES;
exports.MantineProvider = MantineProvider.MantineProvider;
exports.useMantineTheme = MantineProvider.useMantineTheme;
exports.ColorSchemeProvider = ColorSchemeProvider.ColorSchemeProvider;
exports.useMantineColorScheme = ColorSchemeProvider.useMantineColorScheme;
exports.GlobalStyles = GlobalStyles.GlobalStyles;
exports.NormalizeCSS = NormalizeCSS.NormalizeCSS;
exports.getFocusStyles = getFocusStyles.getFocusStyles;
exports.getFontStyles = getFontStyles.getFontStyles;
exports.getSharedColorScheme = getSharedColorScheme.getSharedColorScheme;
exports.getSizeValue = getSizeValue.getSizeValue;
exports.getThemeColor = getThemeColor.getThemeColor;
exports.hexToRgba = hexToRgba.hexToRgba;
exports.useExtractedMargins = useExtractedMargins.useExtractedMargins;
exports.useSx = useSx.useSx;
exports.keyframes = react.keyframes;
exports.createStyles = createStyles.createStyles;
exports.Global = Global.Global;
exports.getCache = CacheProvider.getCache;
exports.useCss = useCss.useCss;
